### PR TITLE
Added close method

### PR DESCRIPTION
**Pull request by lukeburns**
Added close method for https://github.com/karissa/hyperdrive-archive-swarm/issues/2

### CODE REVIEW

**Comment by lukeburns** [comment]
Depends on https://github.com/mafintosh/signalhub/pull/22

File: index.js [hunk 1]
```diff
@@ -58,7 +58,13 @@ module.exports = function (hub, opts) {
     peer.once('close', onclose)
   }
 
-  hub.subscribe('all').pipe(through.obj(function (data, enc, cb) {
+  swarm.close = function (cb) {
+    swarm.emit('close')
+    if (typeof cb === 'function') cb()
```
Comment (mafintosh) [inline index.js]: this should wait for all connections to close before emitting close

File: index.js [hunk 2]
```diff
@@ -98,11 +108,15 @@ module.exports = function (hub, opts) {
     var data = {type: 'connect', from: me}
     data = wrap(data, 'all')
     hub.broadcast('all', data, function () {
-      setTimeout(connect, Math.floor(Math.random() * 2000) + (swarm.peers.length ? 13000 : 3000))
+      var timeout = setTimeout(connect, Math.floor(Math.random() * 2000) + (swarm.peers.length ? 13000 : 3000))
+      swarm.on('close', function () {
+        clearTimeout(timeout)
+      })
```
Comment (mafintosh) [inline index.js]: you need to remove this listener when connect is called

File: index.js [hunk 3]
```diff
@@ -86,6 +92,10 @@ module.exports = function (hub, opts) {
         offerConstraints: offerConstraints
       })
 
+      swarm.on('close', function () {
+        peer.destroy()
+      })
```
Comment (mafintosh) [inline index.js:133]: you need to remove this listener when the peer closes

**Comment by juliangruber** [comment]
@mafintosh in case you missed this commit ^
@lukeburns idk if you pinged @mafintosh yourself, but since github doesn't notify you about new commits, only new comments, it's generally a good idea to comment as well. if you told him, just ignore this :)

**Comment by lukeburns** [comment]
Pinged him on IRC!

File: test.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+var tape = require('tape')
+var signalhub = require('signalhub')
+var swarm = require('./')
+var wrtc = require('electron-webrtc')()
+
+tape('greet and close', function (t) {
+  t.plan(5)
+
+  var peer1 = swarm(signalhub('swarmtest', 'https://signalhub.mafintosh.com'), {wrtc})
```
Comment (mafintosh) [inline test.js]: could you make `swarmtest` a random string? so it doesn't conflict when two people are running the tests

File: index.js [hunk 5]
```diff
@@ -1,69 +1,117 @@
...
+  this.closed = false
+
+  subscribe(this, hub)
+}
+
+inherits(WebRTCSwarm, events.EventEmitter)
+
+WebRTCSwarm.WEBRTC_SUPPORT = SimplePeer.WEBRTC_SUPPORT
+
+WebRTCSwarm.prototype.setup = function (peer, id) {
```
Comment (mafintosh) [inline index.js]: this should probably be _setup since its a private method

File: test.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      t.equal(c.toString(), goodbye, 'goodbye received')
+      peer1.close(function () {
+        t.ok(1, 'peer 1 closed')
+        if (wrtc) wrtc.close()
+      })
+    })
+  })
+
+  peer2.on('peer', function (peer, id) {
+    console.log('peer 2 joined by', id)
```
Comment (mafintosh) [inline test.js]: remote this log or use t.pass(msg)

**Comment by mafintosh** [comment]
Added final comments. @lukeburns have you tested (somehow) that this is still backwards compatible with we had before? I.e. that this PR can connect with webrtc-swarms running the current master?

**Comment by mafintosh** [comment]
@juliangruber i've noticed that github just started notifying on new commits!

**Comment by lukeburns** [comment]
Just tested backwards compatibility by sharing a dat from remote dat.land to a local dat.land running on https://github.com/karissa/hyperdrive-archive-swarm/issues/2 which is running on this.

**Comment by lukeburns** [comment]
Let me run a couple more tests on compatibility

**Comment by mafintosh** [comment]
2.6.0

**Pull request merged**